src/ui/server/controllers/insurance/policy/broker-addresses/save-and-back/index.ts

```diff
@@ -0,0 +1,56 @@
+import { ROUTES } from '../../../../../constants';
+import constructPayload from '../../../../../helpers/construct-payload';
+import generateValidationErrors from '../../../../../shared-validation/yes-no-radios-form';
+import getChosenOrdnanceSurveyAddress from '../../../../../helpers/get-chosen-ordnance-survey-address';
+import mapAndSave from '../../map-and-save/broker';
+import { FIELD_ID, ERROR_MESSAGE } from '..';
+import api from '../../../../../api';
+import { Request, Response } from '../../../../../../types';
+
+const { INSURANCE_ROOT, ALL_SECTIONS, PROBLEM_WITH_SERVICE } = ROUTES.INSURANCE;
+
+/**
+ * saves and goes back to all sections from broker addresses page unless there are API errors
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.redirect} redirects to all sections page on success
+ */
+const post = async (req: Request, res: Response) => {
+  try {
+    const { application } = res.locals;
+
+    if (!application) {
+      return res.redirect(PROBLEM_WITH_SERVICE);
+    }
+
+    const { broker, referenceNumber } = application;
+
+    const payload = constructPayload(req.body, [FIELD_ID]);
+
+    const validationErrors = generateValidationErrors(payload, FIELD_ID, ERROR_MESSAGE);
+
+    if (!validationErrors) {
+      const { postcode, buildingNumberOrName } = broker;
+
+      const response = await api.keystone.getOrdnanceSurveyAddresses(String(postcode), String(buildingNumberOrName));
+
+      const { addresses } = response;
```

Comment:
This will fail if `response` is false, do we want to validate `response`?

Reply:
No need to 😎 

1. `api.keystone.getOrdnanceSurveyAddresses` checks the API response and returned data. If no data is returned (which will contain `addresses`, `api.keystone.getOrdnanceSurveyAddresses` will throw an error.
  1a. Can see all the checks from [this file](https://github.com/UK-Export-Finance/exip/blob/main-broker-address-lookup/src/ui/server/api/keystone/get-ordnance-survey-addresses/index.ts#L20)
3. If any of the above errors occur, `broker-addresses/save-and-back` will in turn, fall into it's own try/catch handler.
4. So if the API response fails, it won't get to line 37 where it tries to obtain `addresses`.

By doing it this way, we avoid having to check the API response object in multiple controllers and instead, it's handled in the UI's DRY API call.